src/actions/editThought.ts

```diff
@@ -134,10 +134,12 @@ const editThought = (state: State, { cursorOffset, force, oldValue, newValue, pa
     [oldKey]: newOldLexeme,
     [newKey]: lexemeNew,
   }
-
-  const sortPreference = getSortPreference(state, editedThought.parentId)
+  const isNote = parentOfEditedThought.value === '=note'
+  const parentThoughtId = isNote
+    ? (getThoughtById(state, parentOfEditedThought.parentId)?.parentId ?? editedThought.parentId)
+    : editedThought.parentId
+  const sortPreference = getSortPreference(state, parentThoughtId)
```

Comment:
Reusing the existing `sortPreference` in the outer scope will incorrectly cause the rank of `thoughtNew` to change when editing a note value within an alphabetically sorted list.

```
- a
- =sort
  - Alphabetical
    - Asc
  - b
    - =note
      - value
```

This may not have any adverse effect, as note values are expected to be the only child of `=note`, so the rank doesn't matter. However, this modification certainly isn't the intended behavior.

Instead, access the sort preference of the note's parent independently using a local variable within the `isNote` block.